src/parsers/MetadataParser.ts

```diff
@@ -95,29 +115,44 @@ export default class MetadataParser extends StatefulBufferParser {
     const mapMetadata = this.parseEncodedMapMetaString(
       this.decodeGameMetaString(encodedString),
     );
-    this.skip(12);
+    const playerCount = this.readUInt32LE();
+    const gameType = this.readStringOfLength(4, "hex");
+    const localeHash = this.readStringOfLength(4, "hex");
     const playerListFinal = playerRecords.concat(
       playerRecords,
       this.parsePlayerList(),
     );
     let reforgedPlayerMetadata: ReforgedPlayerMetadata[] = [];
-    if (this.readUInt8() !== 25) {
-      this.skip(-1);
+    if (this.peekUInt8() !== 25) {
       reforgedPlayerMetadata = this.parseReforgedPlayerMetadata();
     }
-    this.skip(2);
+    if (this.readUInt8() !== 25) {
+      console.error("Unknown chunk detected!");
+      console.error(this.buffer.subarray(this.getOffset() - 1));
+    }
+    const remainingBytes = this.readUInt16LE();
     const slotRecordCount = this.readUInt8();
+    // remaining bytes are: slotRecordCount(1), slots(9*count), seed(4), mode(1), spots(1)
+    if (remainingBytes !== 1 + slotRecordCount * 9 + 6) {
+      console.error(
+        `Remaining bytes (${remainingBytes}) do not match expected bytes (${1 + slotRecordCount * 9 + 6})`,
```

Comment:
You are asserting console.log not to have been called in the test, but you use .error here. Can we match the error outputs to use console.log because thats how the lib currently logs all errors. I am aware that this is not good, and I am planning to change that in the future.